shared/util/codeql/util/test/InlineExpectationsTest.qll

```diff
@@ -779,22 +779,33 @@ module TestPostProcessing {
       )
     }
 
-    private string getTagRegex() {
-      exists(string sourceSinkTags |
-        (
-          getQueryKind() = "problem"
-          or
-          not exists(getSourceTag(_)) and
-          not exists(getSinkTag(_))
-        ) and
-        sourceSinkTags = ""
-        or
-        sourceSinkTags = "|" + getSourceTag(_) + "|" + getSinkTag(_)
+    bindingset[x, y]
+    private int exactDivide(int x, int y) { x % y = 0 and result = x / y }
+
+    /** Gets the `n`th related location selected in `row`. */
+    private TestLocation getRelatedLocation(int row, int n, string element) {
+      n >= 0 and
+      exists(int column |
+        mainQueryResult(row, column, result) and
+        queryResults(mainResultSet(), row, column + 1, element)
       |
-        result = "(Alert" + sourceSinkTags + ")(\\[(.*)\\])?"
+        getQueryKind() = "path-problem" and
+        n = exactDivide(column - 8, 3)
+        or
+        getQueryKind() = "problem" and
+        n = exactDivide(column - 3, 3)
```

Comment:
It's hard to figure out where these numbers come from, can you add an inline explanation to each line?

Reply:
Good call. I added a clarifying comment and noticed a bug, namely that the 8 should have been a 7.

There was no `path-problem` query using related locations, so it didn't get caught initially. I went ahead and added some better tests for this as well.